datafusion/sql/src/unparser/expr.rs

```diff
@@ -1831,15 +1625,15 @@ mod tests {
             ),
             (
                 Expr::Literal(ScalarValue::Date64(Some(0))),
-                r#"CAST('1970-01-01 00:00:00' AS DATETIME)"#,
+                r#"CAST('1970-01-01 00:00:00' AS TIMESTAMP)"#,
```

Comment:
This is still an open issue and we need to check whether this is equivalent. 

Maybe I also misunderstood `Date64` (see `LogicalScalar::from`)

Reply:
dateTime and timestamp is not equivalent

Reply:
Ok then we would need to separate `Date32` and `Date64` which you mentioned in another comment. I only have `Date32` in this PR as there is only `NativeType::Date` which mentions the definition of `Date32`.